src/VecSim/algorithms/svs/svs.h

```diff
@@ -0,0 +1,454 @@
+                                VecSimQueryParams *queryParams) const override {
+        auto rep = new VecSimQueryReply(this->allocator);
+        this->lastMode = STANDARD_KNN;
+        if (k == 0 || this->indexSize() == 0) {
+            return rep;
+        }
+
+        // limit result size to index size
+        k = std::min(k, this->indexSize());
+
+        auto processed_query_ptr = this->preprocessQuery(queryBlob);
+        const void *processed_query = processed_query_ptr.get();
+
+        auto queries = svs::data::ConstSimpleDataView<DataType>{
+            reinterpret_cast<const DataType *>(processed_query), 1, params_.dim};
+        auto result = svs::QueryResult<size_t>{queries.size(), k};
+        auto sp = details::joinSearchParams(impl_->get_search_parameters(), queryParams);
+
+        auto timeoutCtx = queryParams ? queryParams->timeoutCtx : nullptr;
+        auto cancel = [timeoutCtx]() { return VECSIM_TIMEOUT(timeoutCtx); };
+
+        impl_->search(result.view(), queries, sp, cancel);
+        if (cancel()) {
+            rep->code = VecSim_QueryReply_TimedOut;
+            return rep;
+        }
+
+        assert(result.n_queries() == 1);
+
+        rep->results.reserve(result.n_neighbors());
+
+        for (size_t i = 0; i < result.n_neighbors(); i++) {
+            rep->results.emplace_back(result.index(0, i), toVecSimDistance(result.distance(0, i)));
+        }
+        return rep;
+    }
+
+    VecSimQueryReply *rangeQuery(const void *queryBlob, double radius,
+                                 VecSimQueryParams *queryParams) const override {
+        auto rep = new VecSimQueryReply(this->allocator);
+        this->lastMode = RANGE_QUERY;
+        if (radius == 0 || this->indexSize() == 0) {
+            return rep;
+        }
+
+        auto timeoutCtx = queryParams ? queryParams->timeoutCtx : nullptr;
+        auto cancel = [timeoutCtx]() { return VECSIM_TIMEOUT(timeoutCtx); };
+
+        auto sp = details::joinSearchParams(impl_->get_search_parameters(), queryParams);
+        const size_t batch_size = queryParams && queryParams->batchSize
+                                      ? queryParams->batchSize
+                                      : sp.buffer_config_.get_search_window_size();
+        // Base search parameters for the iterator schedule.
+        auto schedule = svs::index::vamana::DefaultSchedule{sp, batch_size};
+
+        auto processed_query_ptr = this->preprocessQuery(queryBlob);
+        const void *processed_query = processed_query_ptr.get();
+
+        std::span<const data_type> query{reinterpret_cast<const data_type *>(processed_query),
+                                         params_.dim};
+        svs::index::vamana::BatchIterator<impl_type, data_type> svs_it{*impl_, query, schedule,
+                                                                       cancel};
+        if (cancel()) {
+            rep->code = VecSim_QueryReply_TimedOut;
+            return rep;
+        }
+
+#if 1
```

Comment:
remove redundant `#if 1`

Reply:
There is the another possible logic for range search which is more accurate but slower. This alternative code is below and disabled now by the `#else` directive.
Can you please review alternatives and let me know which one should be kept - so I will remove the another one.

Reply:
Cleaned-up and fixed.